pkg/kubelet/kuberuntime/kuberuntime_sandbox_test.go

```diff
@@ -85,7 +88,10 @@ func TestCreatePodSandbox_RuntimeClass(t *testing.T) {
 			id, _, err := m.createPodSandbox(pod, 1)
 			if test.expectError {
 				assert.Error(t, err)
-				assert.Contains(t, fakeRuntime.Called, "RunPodSandbox")
+				// When the runtime class is not found, the RunPodSandbox is not called and createPodSandbox return early.
```

Comment:
I think the test is wrongly constructed, b/c there are more than just this error that might lead to `LookupRuntimeHandler` returning error thus not running `RunPodSandbox`. I'd honestly prefer these tests be more thought-out.

Reply:
The current fix is reasonable for the time being, but it would be nice to add some TODO there, please.

Reply:
agree, i don't understand the `defer` logic there and also I suspect there might be some flake in the feature gate logic.

Reply:
yeah, there is a `hack/verify-test-featuregates.sh` that insist on it being deferred... but I still don't get it... :confused:

Reply:
oh i get it now. `SetFeatureGateDuringTest` sets the feature gate and returns a function that will unset it at the end.